packages/beacon-node/src/chain/regen/regen.ts

```diff
@@ -108,26 +118,31 @@ export class StateRegenerator implements IStateRegeneratorInternal {
       });
     }
 
-    const latestCheckpointStateCtx = this.modules.checkpointStateCache.getLatest(blockRoot, computeEpochAtSlot(slot));
+    const {checkpointStateCache} = this.modules;
+    const getLatestApi = shouldReload
+      ? checkpointStateCache.getOrReloadLatest.bind(checkpointStateCache)
+      : checkpointStateCache.getLatest.bind(checkpointStateCache);
+    const latestCheckpointStateCtx = await getLatestApi(blockRoot, computeEpochAtSlot(slot));
```

Comment:
`getLatest` does not return a promise

```suggestion
const latestCheckpointStateCtx = shouldReload
      ? await checkpointStateCache.getOrReloadLatest(blockRoot, computeEpochAtSlot(slot))
      : checkpointStateCache.getLatest(blockRoot, computeEpochAtSlot(slot));
```